state/execution_test.go

```diff
@@ -404,3 +404,111 @@ func TestEndBlockValidatorUpdatesResultingInEmptySet(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.NotEmpty(t, state.NextValidators.Validators)
 }
+
+func TestCheckBlockWithAppError(t *testing.T) {
+	app := &testApp{}
```

Comment:
lots of duplicate code for those 3 tests. can we abstract them away?

Reply:
integrated into one test now